README.md

```diff
@@ -1,17 +1,33 @@
+  - If you are a modder and are interested in a data generator for your textures, you should read the documentation found in `EnchantedBookModelProvider`.
+- If a model does not exist for a registered enchantment when models are baked, then your enchantment is simply ignored and the base `minecraft:item/enchanted_book` is used instead. There is no override or fake model, the vanilla model is used directly.
+  - If there are any missing models for enchantments, a warning will be displayed to the console log for debugging purposes.
 
-And, that's it. The new texture is now successfully added. Don't worry: If the according texture cannot be found, it will just default to the vanilla texture.
+If you are interested in learning more about how our custom item model system works, you can read the documentation found in `EnchantedBookOverrides`.
+
+## Adding Native Compatibility
+
+To add native compatibility for a mod in NEBs itself, the process is mostly the same. Make sure the texture exists in the correct path in `assets/nebs/textures/item`, run the data generator with `runData`, and the accompanying model files should now exist.
+
+## Technical Details for Nerds
+
+This mod uses a special extension of `ItemOverrides` that gives us complete control over what baked models are sent to the renderer. What normally happens in `ItemOverrides` is the item property functions are run through and a baked model for the specified override is given back to the player. However, since the overrides instance already has access to the item stack, and property functions usually query the item stack to begin with to get their desired value, we can easily override this functionality to allow us to work with the enchantment directly and load whatever model we want as a result.
+
+To do this, we have to do our own model baking inside of `EnchantedBookOverrides`. This happens in roughly the same way model baking happens in the base `ItemOverrides` class, but the difference is we apply our baking on top of it so we can use our own functionality without destroying how item rendering works. There are javadocs and comments littered throughout the mod detailing this process, so if it interests you, feel free to take a look. There is unfortunately no Forge event to supply your own `ItemOverrides` instance for an existing model, so we use an injector in `BlockModelMixin` to inject our own overrides instance if the model the game is getting it for is `minecraft:item/enchanted_book`.
 
 > [!NOTE]
-> If you run this mod in your development environment, remember to always run `runData`, when you changed the [properties.json](./common/main/resources/assets/nebs/models/properties.json), before doing `runClient`. It will automatically add all the required info to `enchanted_book.json` in addition to the necessary json files.

```

Comment:
Please keep this note for people contributing to this project.

Reply:
Oops, didn't mean to delete that one.